src/Compilers/CSharp/Portable/BoundTree/BoundNode.cs

```diff
@@ -100,6 +100,20 @@ public SyntaxTree SyntaxTree
             }
         }
 
+        protected void CopyAttributes(BoundNode original)
+        {
+            if (this.WasCompilerGenerated)
+            {
+                this.WasCompilerGenerated = original.WasCompilerGenerated;
+            }
+
+            Debug.Assert(original is BoundExpression || !original.IsSuppressed);
+            if (original.IsSuppressed)
+            {
+                this.IsSuppressed = original.IsSuppressed;
+            }
```

Comment:
I realize each attribute checks the value is not already set, but ignoring that, it seems strange that the contract of this method is to only set the attributes if the original attributes were set.

Consider setting the attributes unconditionally:
```C#
this.WasCompilerGenerated = original.WasCompilerGenerated;
this.IsSuppressed = original.IsSuppressed;
```

Reply:
What about VB?
I tried to align the two implementations since it is possible.

---
In reply to: [252046440](https://github.com/dotnet/roslyn/pull/32850#discussion_r252046440) [](ancestors = 252046440)

Reply:
The two implementations will be distinct since VB uses `SetWasCompilerGenerated()` that does not allow setting `False`.

---
In reply to: [252049465](https://github.com/dotnet/roslyn/pull/32850#discussion_r252049465) [](ancestors = 252049465,252046440)

Reply:
The C# `WasCompilerGenerated` setter also does not allow setting to false

---
In reply to: [252051070](https://github.com/dotnet/roslyn/pull/32850#discussion_r252051070) [](ancestors = 252051070,252049465,252046440)